Segmentation violation from `nsc import account`

Segmentation violation from `nsc import account` when no nsc environment has been set up.  On nsc branch main.   The error occurs before the specified jwt is checked for validity, so passing any file will cause the error.

```bash
root@instance:/# nsc import account --file $HOME/client_account.jwt
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4b6c58]

goroutine 1 [running]:
github.com/nats-io/nsc/v2/cmd/store.(*Store).IsManaged(...)
	/nsc/cmd/store/store.go:222
github.com/nats-io/nsc/v2/cmd.(*SignerParams).SetDefaults(0x4000157810, 0x70, 0x1?, {0x6e7be0?, 0x40001f2b70?})
	/nsc/cmd/signerparams.go:40 +0xd8
github.com/nats-io/nsc/v2/cmd.(*ImportAccount).SetDefaults(0x40001ee720?, {0x6e7be0?, 0x40001f2b70?})
	/nsc/cmd/importaccount.go:107 +0x38
github.com/nats-io/nsc/v2/cmd.run({0x6e7be0, 0x40001f2b70}, {0x5ab740?, 0x4000157810?})
	/nsc/cmd/action.go:120 +0x60
github.com/nats-io/nsc/v2/cmd.RunMaybeStorelessAction(0x40001ce900, {0x4000063b00?, 0x0, 0x2}, {0x5ab740, 0x4000157810})
	/nsc/cmd/action.go:95 +0x100
github.com/nats-io/nsc/v2/cmd.createImportAccountCmd.func1(0x40001ce900?, {0x4000063b00?, 0x2?, 0x2?})
	/nsc/cmd/importaccount.go:39 +0x2c
github.com/spf13/cobra.(*Command).execute(0x40001ce900, {0x4000063ae0, 0x2, 0x2})
	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x5e0
github.com/spf13/cobra.(*Command).ExecuteC(0xa5ae00)
	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x368
github.com/spf13/cobra.(*Command).Execute(...)
	/root/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/nats-io/nsc/v2/cmd.ExecuteWithWriter({0x6e23c0?, 0x4000010028?})
	/nsc/cmd/root.go:228 +0x6c
github.com/nats-io/nsc/v2/cmd.Execute()
	/nsc/cmd/root.go:216 +0x40
main.main()
	/nsc/main.go:34 +0xd4
```

--- a/cmd/importaccount.go
+++ b/cmd/importaccount.go
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018-2022 The NATS Authors
+ * Copyright 2018-2023 The NATS Authors
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
@@ -83,7 +83,11 @@ func (p *fileImport) Load(ctx ActionCtx) (err error) {
 }
 
 func (p *fileImport) Validate(xtx ActionCtx, fileEndings ...string) error {
-	fi, err := os.Stat(p.file)
+	fp, err := Expand(p.file)
+	if err != nil {
+		return err
+	}
+	fi, err := os.Stat(fp)
 	if err != nil {
 		return err
 	}

--- a/cmd/store/store.go
+++ b/cmd/store/store.go
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018-2022 The NATS Authors
+ * Copyright 2018-2023 The NATS Authors
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
@@ -46,6 +46,12 @@ const Accounts = "accounts"
 var standardDirs = []string{Accounts}
 
 var ErrNotExist = errors.New("resource does not exist")
+
+const NoStoreSetError = "no store set"
+
+var ErrNoStoreSet = errors.New(NoStoreSetError)
+
+const Empty = ""
 
 // Store is a directory that contains nsc assets
 type Store struct {
@@ -172,7 +178,7 @@ func CreateStore(env string, operatorsDir string, operator *NamedKey) (*Store, e
 	}
 
 	for _, d := range standardDirs {
-		dp := s.resolve(d, "")
+		dp := s.Resolve(d, Empty)
 		if err = os.MkdirAll(dp, 0700); err != nil {
 			return nil, fmt.Errorf("error creating %#q: %v", dp, err)
 		}
@@ -219,20 +225,22 @@ func LoadStore(dir string) (*Store, error) {
 }
 
 func (s *Store) IsManaged() bool {
+	if s == nil {
+		return false
+	}
 	return s.Info.Managed
 }
 
-func (s *Store) resolve(name ...string) string {
-	return filepath.Join(s.Dir, filepath.Join(name...))
-}
-
 func (s *Store) Resolve(name ...string) string {
+	if s == nil {
+		return Empty
+	}
 	return filepath.Join(s.Dir, filepath.Join(name...))
 }
 
 // Has returns true if the specified asset exists
 func (s *Store) Has(name ...string) bool {
-	fp := s.resolve(name...)
+	fp := s.Resolve(name...)
 	return s.has(fp)
 }
 
@@ -249,9 +257,12 @@ func (s *Store) has(fp string) bool {
 
 // Read reads the specified file name or subpath from the store
 func (s *Store) Read(name ...string) ([]byte, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	s.Lock()
 	defer s.Unlock()
-	fp := s.resolve(name...)
+	fp := s.Resolve(name...)
 	d, err := os.ReadFile(fp)
 	if err != nil {
 		return nil, fmt.Errorf("error reading %#q: %w", fp, err)
@@ -261,10 +272,13 @@ func (s *Store) Read(name ...string) ([]byte, error) {
 
 // Write writes the specified file name or subpath in the store
 func (s *Store) Write(data []byte, name ...string) error {
+	if s == nil {
+		return ErrNoStoreSet
+	}
 	s.Lock()
 	defer s.Unlock()
 
-	fp := s.resolve(name...)
+	fp := s.Resolve(name...)
 	dp := filepath.Dir(fp)
 
 	if err := os.MkdirAll(dp, 0700); err != nil {
@@ -274,22 +288,31 @@ func (s *Store) Write(data []byte, name ...string) error {
 }
 
 func (s *Store) List(path ...string) ([]fs.DirEntry, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	s.Lock()
 	defer s.Unlock()
 
-	fp := s.resolve(path...)
+	fp := s.Resolve(path...)
 	return os.ReadDir(fp)
 }
 
 // Delete the specified file name or subpath from the store
 func (s *Store) Delete(name ...string) error {
+	if s == nil {
+		return ErrNoStoreSet
+	}
 	s.Lock()
 	defer s.Unlock()
-	fp := s.resolve(name...)
+	fp := s.Resolve(name...)
 	return os.Remove(fp)
 }
 
 func (s *Store) ListSubContainers(name ...string) ([]string, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	var containers []string
 	fp := filepath.Join(name...)
 	if s.Has(fp) {
@@ -309,6 +332,9 @@ func (s *Store) ListSubContainers(name ...string) ([]string, error) {
 }
 
 func (s *Store) ListEntries(name ...string) ([]string, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	var entries []string
 	fp := filepath.Join(name...)
 	if s.Has(fp) {
@@ -412,6 +438,9 @@ func (s *Store) handleManagedAccount(data []byte) (*Report, error) {
 }
 
 func (s *Store) StoreClaim(data []byte) (*Report, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	ct, err := s.ClaimType(data)
 	if err != nil {
 		return nil, err
@@ -449,6 +478,9 @@ func (s *Store) StoreClaim(data []byte) (*Report, error) {
 }
 
 func (s *Store) StoreRaw(data []byte) error {
+	if s == nil {
+		return ErrNoStoreSet
+	}
 	ct, err := s.ClaimType(data)
 	if err != nil {
 		return err
@@ -505,6 +537,9 @@ func (s *Store) StoreRaw(data []byte) error {
 }
 
 func (s *Store) GetName() string {
+	if s == nil {
+		return Empty
+	}
 	return s.Info.Name
 }
 
@@ -524,6 +559,9 @@ func PlainName(name string) string {
 }
 
 func (s *Store) loadJson(v interface{}, name ...string) error {
+	if s == nil {
+		return ErrNoStoreSet
+	}
 	if s.Has(name...) {
 		d, err := s.Read(name...)
 		if err != nil {
@@ -539,6 +577,9 @@ func (s *Store) loadJson(v interface{}, name ...string) error {
 }
 
 func (s *Store) LoadClaim(name ...string) (*jwt.GenericClaims, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	if s.Has(name...) {
 		d, err := s.Read(name...)
 		if err != nil {
@@ -566,6 +607,9 @@ func (s *Store) ReadOperatorClaim() (*jwt.OperatorClaims, error) {
 }
 
 func (s *Store) ReadRawOperatorClaim() ([]byte, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	fn := JwtName(s.GetName())
 	if s.Has(fn) {
 		d, err := s.Read(fn)
@@ -590,6 +634,9 @@ func (s *Store) ReadAccountClaim(name string) (*jwt.AccountClaims, error) {
 }
 
 func (s *Store) ReadRawAccountClaim(name string) ([]byte, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	if s.Has(Accounts, name, JwtName(name)) {
 		d, err := s.Read(Accounts, name, JwtName(name))
 		if err != nil {
@@ -613,6 +660,9 @@ func (s *Store) ReadUserClaim(accountName string, name string) (*jwt.UserClaims,
 }
 
 func (s *Store) ReadRawUserClaim(accountName string, name string) ([]byte, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	if s.Has(Accounts, accountName, Users, JwtName(name)) {
 		d, err := s.Read(Accounts, accountName, Users, JwtName(name))
 		if err != nil {
@@ -624,6 +674,9 @@ func (s *Store) ReadRawUserClaim(accountName string, name string) ([]byte, error
 }
 
 func (s *Store) LoadRootClaim() (*jwt.GenericClaims, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	fn := JwtName(s.GetName())
 	if s.Has(fn) {
 		return s.LoadClaim(fn)
@@ -632,6 +685,9 @@ func (s *Store) LoadRootClaim() (*jwt.GenericClaims, error) {
 }
 
 func (s *Store) LoadDefaultEntity(kind string) (*jwt.GenericClaims, error) {
+	if s == nil {
+		return nil, ErrNoStoreSet
+	}
 	dirs, err := s.ListSubContainers(kind)
 	if err != nil {
 		return nil, fmt.Errorf("error listing %s: %w", kind, err)
@@ -667,6 +723,9 @@ func (s *Store) LoadDefaultEntity(kind string) (*jwt.GenericClaims, error) {
 }
 
 func (s *Store) GetRootPublicKey() (string, error) {
+	if s == nil {
+		return "", ErrNoStoreSet
+	}
 	c, err := s.LoadRootClaim()
 	if err != nil {
 		return "", err

--- a/cmd/store/store_test.go
+++ b/cmd/store/store_test.go
@@ -1,18 +1,16 @@
 /*
+ * Copyright 2018-2023 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
  *
- *  * Copyright 2018-2019 The NATS Authors
- *  * Licensed under the Apache License, Version 2.0 (the "License");
- *  * you may not use this file except in compliance with the License.
- *  * You may obtain a copy of the License at
- *  *
- *  * http://www.apache.org/licenses/LICENSE-2.0
- *  *
- *  * Unless required by applicable law or agreed to in writing, software
- *  * distributed under the License is distributed on an "AS IS" BASIS,
- *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  * See the License for the specific language governing permissions and
- *  * limitations under the License.
+ * http://www.apache.org/licenses/LICENSE-2.0
  *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  */
 
 package store
@@ -370,4 +368,74 @@ func TestStore_GetAccountKeys(t *testing.T) {
 	require.Len(t, keys, 2)
 	require.Equal(t, apub, keys[0])
 	require.Equal(t, apub2, keys[1])
+}
+
+func assertErrorMessage(t *testing.T, err error, errMsg string) {
+	require.Error(t, err)
+	if errMsg != "" {
+		require.Equal(t, errMsg, err.Error())
+	}
+}
+
+func assertError(t *testing.T, v interface{}, err error, errMsg string) {
+	require.Nil(t, v)
+	assertErrorMessage(t, err, errMsg)
+}
+
+func TestStore_NilStore(t *testing.T) {
+	var s *Store
+	require.Empty(t, s.Resolve("A"))
+	require.False(t, s.IsManaged())
+	require.False(t, s.Has("foo"))
+	require.False(t, s.HasAccount("foo"))
+	require.Equal(t, "", s.GetName())
+	assertErrorMessage(t, s.Write(nil, "hello"), NoStoreSetError)
+	assertErrorMessage(t, s.StoreRaw(nil), NoStoreSetError)
+	assertErrorMessage(t, s.Delete("foo"), NoStoreSetError)
+
+	list, err := s.List(Accounts)
+	assertError(t, list, err, NoStoreSetError)
+
+	r, err := s.StoreClaim(nil)
+	assertError(t, r, err, NoStoreSetError)
+
+	k, err := s.GetRootPublicKey()
+	require.Empty(t, k)
+	assertErrorMessage(t, err, NoStoreSetError)
+
+	e, err := s.ListEntries("foo")
+	assertError(t, e, err, NoStoreSetError)
+
+	sub, err := s.ListSubContainers("A")
+	assertError(t, sub, err, NoStoreSetError)
+
+	cd, err := s.LoadClaim("A")
+	assertError(t, cd, err, NoStoreSetError)
+
+	gc, err := s.LoadDefaultEntity(Accounts)
+	assertError(t, gc, err, NoStoreSetError)
+
+	gc, err = s.LoadRootClaim()
+	assertError(t, gc, err, NoStoreSetError)
+
+	ra, err := s.Read("A")
+	assertError(t, ra, err, NoStoreSetError)
+
+	ac, err := s.ReadAccountClaim("A")
+	assertError(t, ac, err, NoStoreSetError)
+
+	d, err := s.ReadRawAccountClaim("A")
+	assertError(t, d, err, NoStoreSetError)
+
+	oc, err := s.ReadOperatorClaim()
+	assertError(t, oc, err, NoStoreSetError)
+
+	d, err = s.ReadRawOperatorClaim()
+	assertError(t, d, err, NoStoreSetError)
+
+	uc, err := s.ReadUserClaim("A", "U")
+	assertError(t, uc, err, NoStoreSetError)
+
+	d, err = s.ReadRawUserClaim("A", "U")
+	assertError(t, d, err, NoStoreSetError)
 }

